app/src/app/components/transformation-view/transformation-view.component.ts

```diff
@@ -117,6 +117,25 @@ export class TransformationViewComponent implements OnInit, OnDestroy {
         });
     }
 
+    downloadScript() {
+        const folder = this.csarId + '-' + this.platform;
+        let result = '#!/bin/bash\n';
+        result += 'timestamp=$(date +%s)\n';
+        result += 'folder=' + folder + '-$timestamp\n';
+        result += 'mkdir $folder\n';
+        result += 'cd $folder\n';
+        result += 'curl ' + this.url + ' --output $folder.csar\n';
+        result += 'unzip $folder.csar -d unzipped\n';
+        result += 'bash unzipped/output/scripts/deploy.sh\n';
+        result += 'cd ../\n';
```

Comment:
Automatic cleanup after script execution...hm. Not sure if this is user-friendly. What if user aborts the execution of the deploy.sh and wants to execute it (e.g., with different parameters?) 
My idea is to prompt the user for cleanup: `cleanup [y/N]?`
Maybe also prompt for execution of deploy.sh. so user has the chance to modify the script.

Reply:
This might be a good idea going to add this tomorrow